drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -65,6 +66,79 @@ local device_type_profile_map = {
   [ON_OFF_SWITCH_ID] = "switch-binary",
   [ON_OFF_DIMMER_SWITCH_ID] = "switch-level",
   [ON_OFF_COLOR_DIMMER_SWITCH_ID] = "switch-color-level",
+  [GENERIC_SWITCH_ID] = "button"
+}
+
+local device_type_attribute_map = {
```

Comment:
Did you consider mapping capabilities to the different attributes/events instead of device types? Then you would know which attributes to subscribe to based on the profile the device is using which seems more flexible.

Reply:
Since the child devices are not created at `init` for the parent device, we don't know their profile until they are initialized, so I was doing the subscriptions for child devices in `device_init` using the `get_parent_device` function. But, after talking to Cooper about this, he pointed out that this could lead to race conditions as noted here: https://github.ecodesamsung.com/iot-hub/scripting-engine/blob/00a4fcdba1d29aa9d3b033909a6aaa1df24cc384/lua_libs/st/device.lua#L217

I can't think of another way to do this outside of `device_init`, so I'm thinking it might be best to continue using the device type.